lib/measured/conversion.rb

```diff
@@ -11,50 +11,43 @@ def initialize(base_unit, units, case_sensitive: false)
   attr_reader :base_unit, :units
 
   def unit_names_with_aliases
-    @units.map{|u| u.names}.flatten.sort
+    @unit_names_with_aliases ||= @units.flat_map(&:names).sort
   end
 
   def unit_names
-    @units.map{|u| u.name}.sort
+    @unit_names ||= @units.map(&:name).sort
   end
 
   def unit_or_alias?(name)
-    @units.each{|unit| return true if unit.names_include?(name, case_sensitive: @case_sensitive)}
-    false
+    @units.any? { |unit| unit.names_include?(name, case_sensitive: @case_sensitive) }
   end
 
   def unit?(name)
-    @units.each{|unit| return true if unit.name_eql?(name, case_sensitive: @case_sensitive)}
-    false
+    @units.any? { |unit| unit.name_eql?(name, case_sensitive: @case_sensitive) }
   end
 
   def to_unit_name(name)
     unit_for(name).name
   end
 
   def convert(value, from:, to:)
-    raise Measured::UnitError, "Source unit #{ from } does not exits." unless unit?(from)
-    raise Measured::UnitError, "Converted unit #{ to } does not exits." unless unit?(to)
```

Comment:
We're not testing these anywhere, so instead of incurring the cost of `unit?` and `unit_for`, I figure we just do `unit_for` and have it raise.